### PR TITLE
Python exports

### PR DESCRIPTION
**Pull request by mbuerki**

### CODE REVIEW

**Comment by HannesSommer** [comment]
add to whitelist

**Comment by HannesSommer** [comment]
add to whitelist

**Comment by mbuerki** [comment]
@HannesSommer in principle, I think this could be merged. any objections?

**Comment by mbuerki** [comment]
@HannesSommer hm, this still depends on numpy_eigen and python_module from schweizer messer.
no idea how to handle this at this point. do you have any suggestions?

**Comment by mbuerki** [comment]
@HannesSommer 
hm I actually prefer the second option. 
I think it could make sense to have the python export feature in separate repository. something like `catkin_python`? or `catkin_python_export`? also for other projects I could imagine...
wdyt?

**Comment by HannesSommer** [comment]
Sorry about my mistake regarding the doc string for the constructor. I've just fixed it.

File: minkindr_python/src/export-rotation-quaternion.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  class_< Quaternion, boost::shared_ptr<Quaternion> >( "Quaternion", init<>() )
+    .def(init<const Eigen::Matrix3d&>())
+    .def(init<const double, const double, const double, const double>())
+    .def("w", &Quaternion::w)
+    .def("x", &Quaternion::x)
+    .def("y", &Quaternion::y)
+    .def("z", &Quaternion::z)
+    .def("getRotationMatrix", &Quaternion::getRotationMatrix)
+    .def("getQuaternionXYZW", getQuaternionXYZW)
```
[the remark this answers is not in the document]
Comment (mbuerki) [reply minkindr_python/src/export-rotation-quaternion.cc]: done

**Comment by mbuerki** [comment]
@HannesSommer your comments should be addressed (let me know if there's anything left to do). 
jenkins fails because I guess it's missing the dependency on the new repos (numpy_eigen, and the boost_python_export... repo). any ideas how to add these deps?

File: minkindr/include/kindr/minimal/implementation/angle-axis-inl.h [hunk 2]
```diff
@@ -64,12 +64,15 @@ AngleAxisTemplate<Scalar>::AngleAxisTemplate(const RotationMatrix& matrix) :
 
 template<typename Scalar>
 AngleAxisTemplate<Scalar>::AngleAxisTemplate(const Vector3& angleAxis) {
-    Scalar norm = angleAxis.norm();
-    CHECK_GT(norm, 1e-8);
-    Vector3 axis = angleAxis;
-    axis = axis / norm;
-
-    C_A_B_ = Implementation(norm, axis);
+    const Scalar angle_rad = angleAxis.norm();
```
Comment (HannesSommer) [inline minkindr/include/kindr/minimal/implementation/angle-axis-inl.h]: This parameter should also be called `rotationVector` (more correct and consistent with the new getter).

**Comment by mbuerki** [comment]
@HannesSommer this should now only contain the python related additions, and all comments should be addressed.

File: minkindr_python/src/export-rotation-quaternion.cc [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#include <glog/logging.h>
+#include <kindr/minimal/rotation-quaternion.h>
+#include <numpy_eigen/boost_python_headers.hpp>
+
+using namespace boost::python;
+
+typedef kindr::minimal::RotationQuaternionTemplate<double> Quaternion;
+
+Eigen::Vector4d getQuaternionXYZW(const Quaternion* quaternion) {
+  return CHECK_NOTNULL(quaternion)->toImplementation().coeffs();
```
Comment (HannesSommer) [inline minkindr_python/src/export-rotation-quaternion.cc]: I overlooked this before, but I would not rely on internal ordering if possible. It is  not even specified in the Eigen doc (at least not prominently). Hence, I would just copy from below (the usage of w() ..) and reorder for safety. 

Btw: you should be able to use const ref for the quaternion and safe the NOTNULL check (same below).

**Pull request merged**